### ISSUE TITLE
use case: using zig cc to cross compile a cgo program for windows

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#7874]
Dowstream issue: https://github.com/golang/go/issues/43886

main.go

```go
package main

//int Add(int a, int b){
//    return a+b;
//}
import "C"
import "fmt"

func main() {
	a := C.int(10)
	b := C.int(20)
	c := C.Add(a, b)
	fmt.Println(c) // 30
}
```

`CGO_ENABLED=1 GOOS=windows GOARCH=amd64 CC="zcc" CXX="zc++" go build main.go`

zcc
```sh
#!/bin/sh
zig cc -target x86_64-windows-gnu $@
```

zc++
```sh
#!/bin/sh
zig c++ -target x86_64-windows-gnu $@
```

output:

```
# command-line-arguments
warning: unsupported linker arg: --compress-debug-sections=zlib-gnu
```

It produces `main.exe` but when run it segfaults in `runtime.rt0_go()` (called from main()). This is probably an issue with the go runtime, and the next step here will be to file a downstream error report with go.

**Comment by LaPingvino** [comment]
as mentioned in the Go issue, this is probably fixed by @zx2c4's fixes for Windows ARM64, which necessitates LLVM support. it seems like a "did never work before" issue.

**Comment by antonmedv** [comment]
Still not possible?

**Comment by LaPingvino** [comment]
Probably fixed with Go 1.18, which is not released yet. This also fixes the need for a workaround for invoking zig cc / zig c++.

**Comment by motiejus** [comment]
The test program works for me with go1.17 and zig 0.9.0 and when run under wine:

```
motiejus ~/x/go $ go version
go version go1.17 linux/amd64
motiejus ~/x/go $ zig version
0.9.0
motiejus ~/x/go $ wine ./main.exe 
30
motiejus ~/x/go $ 
```
